engine/tests/phpunit/Elgg/TestCase.php

```diff
@@ -19,6 +20,16 @@
 	 */
 	public function __construct($name = NULL, array $data = array(), $dataName = '') {
 		self::bootstrap();
+
+		// force plugins->find() to return []
+		$plugins_mock = $this->getMockBuilder(Plugins::class)
```

Comment:
How about we add `\Elgg\Di\TestingServiceProvider` that extends `\Elgg\Di\ServiceProvider` and override factories for testing purposes? I think that might save us a lot of hassle later?